### PR TITLE
Commerce 1539 (2/7) Initialize providers without using 'this'

### PR DESCRIPTION
**Pull request by alecsloan**

### CODE REVIEW

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
ci:test

**Comment by alecsloan** [comment]
@marco-leo Rebased

**Comment by alecsloan** [comment]
ci:test

File: commerce-cart-taglib/src/main/java/com/liferay/commerce/cart/taglib/servlet/taglib/internal/js/loader/modules/extender/npm/NPMResolverProvider.java [hunk 1]
```diff
@@ -34,7 +35,12 @@ public static NPMResolver getNPMResolver() {
 	}
 
 	public NPMResolverProvider() {
```
Comment (ethanbustad) [inline commerce-cart-taglib/src/main/java/com/liferay/commerce/cart/taglib/servlet/taglib/internal/js/loader/modules/extender/npm/NPMResolverProvider.java:37]: don't think there's any need to declare an empty constructor here.

File: commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/internal/js/loader/modules/extender/npm/NPMResolverProvider.java [hunk 2]
```diff
@@ -34,7 +35,12 @@ public static NPMResolver getNPMResolver() {
 	}
 
 	public NPMResolverProvider() {
```
Comment (ethanbustad) [inline commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/internal/js/loader/modules/extender/npm/NPMResolverProvider.java:37]: same here

File: commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/internal/util/CPContentHelperProvider.java [hunk 3]
```diff
@@ -34,9 +35,14 @@ public static CPContentHelper getCPContentHelper() {
...
 	}
 
+	protected static void setCPContentHelperProvider(
+		CPContentHelperProvider cpContentHelperProvider) {
+
+		_cpContentHelperProvider = cpContentHelperProvider;
+	}
+
 	@Activate
 	protected void activate() {
```
Comment (ethanbustad) [inline commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/internal/util/CPContentHelperProvider.java:45]: no need to have this empty `activate` method any more.

File: commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/internal/util/ProductHelperProvider.java [hunk 4]
```diff
@@ -34,9 +35,14 @@ public static ProductHelper getProductHelper() {
...
 	}
 
+	protected static void setProductHelperProvider(
+		ProductHelperProvider productHelperProvider) {
+
+		_productHelperProvider = productHelperProvider;
+	}
+
 	@Activate
 	protected void activate() {
```
Comment (ethanbustad) [inline commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/internal/util/ProductHelperProvider.java:45]: same here

**Review by ethanbustad — COMMENTED** [review]
@alecsloan a couple things to clean up here.

**Comment by alecsloan** [comment]
@ethanbustad  I actually Remove all of these providers in my new PRs for 2235/2237/2238. The pattern that we're using now is in ServletContextUtil

**Pull request merged**